examples/caching/src/test/kotlin/cnt/cache/domain/MovieAccessorTest.kt

```diff
@@ -0,0 +1,97 @@
+package cnt.cache.domain
+
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import io.quarkiverse.test.junit.mockk.InjectMock
+import io.quarkus.test.junit.QuarkusTest
+import java.io.IOException
+import javax.inject.Inject
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+
+
+private const val MARC_HAMILL = "Marc Hamill"
+private const val FELICITY_JONES = "Felicity Jones"
+private const val CARRIE_FISCHER = "Carrie Fischer"
+
+/**
+ * The correct functionality of our own code is verified using unit-tests.
+ */
+internal class MovieAccessorTest {
+
+    private val client: MovieClient = mockk()
+    private val cut = MovieAccessor(client)
+
+    @Test
+    fun `invokes client and returns number of pages if found`() {
+        every { client.getNumberOfMoviesForActor(MARC_HAMILL) } returns 46
+        val actual = cut.getNumberOfMoviesForActor(MARC_HAMILL)
+        assertThat(actual).isEqualTo(46)
+    }
+
+    @Test
+    fun `invokes invokes client and returns null if null was returned`() {
+        every { client.getNumberOfMoviesForActor(FELICITY_JONES) } returns null
+        val actual = cut.getNumberOfMoviesForActor(FELICITY_JONES)
+        assertThat(actual).isNull()
+    }
+
+    @Test
+    fun `invokes invokes client and rethrow IOExeption`() {
```

Comment:
This test might now be unnecessary. Not handling an exception (anymore) is not really a feature to test.
When we find out how to cache unless the value is `null` the exception handling test might return.